tests/testthat/test-type-unspecified.R

```diff
@@ -77,3 +77,19 @@ test_that("unspecified() validates input", {
 test_that("tibble::type_sum() knows about unspecified", {
   expect_identical(tibble::type_sum(unspecified(3)), "???")
 })
+
+test_that("casting to a scalar type errors", {
+  expect_error(vec_cast(NA, quote(x)), "Not a vector")
+  expect_error(vec_cast(unspecified(1), quote(x)), "Not a vector")
+})
+
+test_that("can cast to unspecified from unspecified", {
```

Comment:
Should this be an error? When do we need to cast to unspecified? I see "unspecified" as a temporary type during the richer type reduction. Once we have the final type, it shouldn't be unspecified anymore, but logical.

Reply:
If it is needed somewhere because of internal type manipulations it is fine to keep it - but then I wonder if we should mark these tests as "monitoring" to distinguish them from definitional tests. I.e. the test doesn't define correct behaviour. I would also mark the slice(uns) tests as monitoring for instance.

Reply:
I think we definitely need `vec_cast(NA, NA)` to work, and at the `vec_typeof()` level that looks identical to `vec_cast(unspecified(1), unspecified(1))`. Both deal with `vctrs_type_unspecified` inputs.

So we can't just error outright.

Right now we have:

```r
vec_cast(unspecified(1), unspecified(1)) == unspecified(1)
vec_cast(NA, unspecified(1))             == unspecified(1)
vec_cast(NA, NA)                         == NA
vec_cast(unspecified(1), NA)             == NA
```

I could make it so that if both `x` and `to` are unspecified, we always return `NA`? Meaning we would call `vec_init(vctrs_shared_empty_lgl, vec_size(x))`. That way we end up with a finalised logical?

Reply:
It's fine if it works just because of an implementation detail, it's just strange to explicitly test for it rather than testing for the actual behaviour of interest. It gives the wrong signal to the reader of the test. It doesn't seem important to monitor this behaviour, but if you'd like to keep these tests, mark them as "monitoring tests"?